sql-db/narayana-transactions/src/test/java/io/quarkus/ts/transactions/TransactionCommons.java

```diff
@@ -0,0 +1,208 @@
+package io.quarkus.ts.transactions;
+
+import static io.restassured.RestAssured.given;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.JaegerService;
+import io.quarkus.test.services.JaegerContainer;
+import io.restassured.http.ContentType;
+import io.restassured.response.Response;
+
+public abstract class TransactionCommons {
+
+    static final String ACCOUNT_NUMBER_MIGUEL = "SK0389852379529966291984";
+    static final String ACCOUNT_NUMBER_GARCILASO = "FR9317569000409377431694J37";
+    static final String ACCOUNT_NUMBER_LUIS = "ES8521006742088984966816";
+    static final String ACCOUNT_NUMBER_LOPE = "CZ9250512252717368964232";
+    static final String ACCOUNT_NUMBER_FRANCISCO = "ES8521006742088984966817";
+    static final int ASSERT_SERVICE_TIMEOUT_MINUTES = 1;
+    private Response jaegerResponse;
+
+    @JaegerContainer(useOtlpCollector = true, expectedLog = "\"Health Check state change\",\"status\":\"ready\"")
+    static final JaegerService jaeger = new JaegerService();
+
+    @Tag("QUARKUS-2492")
+    @Test
+    public void verifyNarayanaProgrammaticApproachTransaction() {
+        TransferDTO transferDTO = new TransferDTO();
+        transferDTO.setAccountFrom(ACCOUNT_NUMBER_MIGUEL);
+        transferDTO.setAccountTo(ACCOUNT_NUMBER_LOPE);
+        transferDTO.setAmount(100);
+
+        given()
+                .contentType(ContentType.JSON)
+                .body(transferDTO).post("/transfer/transaction")
+                .then().statusCode(HttpStatus.SC_CREATED);
+
+        AccountEntity miguelAccount = getAccount(ACCOUNT_NUMBER_MIGUEL);
+        Assertions.assertEquals(0, miguelAccount.getAmount(), "Unexpected account amount.");
+
+        AccountEntity lopeAccount = getAccount(ACCOUNT_NUMBER_LOPE);
+        Assertions.assertEquals(200, lopeAccount.getAmount(), "Unexpected account amount.");
+
+        JournalEntity miguelJournal = getLatestJournalRecord(ACCOUNT_NUMBER_MIGUEL);
+        Assertions.assertEquals(100, miguelJournal.getAmount(), "Unexpected journal amount.");
+    }
+
+    @Tag("QUARKUS-2492")
+    @Test
+    public void verifyNarayanaLambdaApproachTransaction() {
+        TransferDTO transferDTO = new TransferDTO();
+        transferDTO.setAccountFrom(ACCOUNT_NUMBER_GARCILASO);
```

Comment:
For these cases I would recommend use null or a special account, but since it is not a real application — that is fine